packages/router/src/util.ts

```diff
@@ -210,9 +210,9 @@ const replaceParams = (path: string, args: Record<string, unknown> = {}) => {
         const paramSpec = part.substr(1, part.length - 2)
         const paramName = paramSpec.split(':')[0]
         const arg = args[paramName]
-        if (arg) {
+        if (Object.prototype.hasOwnProperty.call(args, paramName)) {
```

Comment:
@cannikin made an excellent suggestion to comment this code, so I did.
I also hinted in the comment why Peter's idea/question about using `typeof !== 'undefined'` doesn't work.

Reply:
(and in the future, if someone does try to change this code to use the `typeof` check, or removes the `'' + arg` string conversion the additional tests I added will fail, so we should be safe against accidentally breaking this in the future if someone tries to "simplify" this code)